[BUG] StackOverflowError

**Description**
When running the nexusIQScan task I receive a StackOverflowError with the following stack trace snippet:

Caused by: java.lang.StackOverflowError: (No message provided)
org.sonatype.gradle.plugins.scan.common.DependenciesFinder.fillAllChildDependencies (DependenciesFinder.java:432)
org.sonatype.gradle.plugins.scan.common.DependenciesFinder.fillAllChildDependencies (DependenciesFinder.java:436) org.sonatype.gradle.plugins.scan.common.DependenciesFinder.fillAllChildDependencies (DependenciesFinder.java:436)
org.sonatype.gradle.plugins.scan.common.DependenciesFinder.fillAllChildDependencies (DependenciesFinder.java:436)
...

**To Reproduce**
I'm working to create a minimal reproducer.  From the stack trace, it appears to be a circular dependency of some sort.

**Expected behavior**
A successful scan.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: OS X 14.4.1 (aarch64)
 - Gradle Version: 8.4
 - JVM Version and Flavor:  Amazon Corretto OpenJDK 64-Bit Server VM 17.0.11+9-LTS (mixed mode, sharing)
 - Nexus IQ Server Version: 174
 - Version: 2.8.2

**Additional context**

1. Appears this logic was introduced in 2.8.0.  Version 2.7.0 scans the same project successfully.
2. excludeCompileOnly = true

--- a/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
+++ b/src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java
@@ -429,7 +429,9 @@ public class DependenciesFinder
   }
 
   private void fillAllChildDependencies(ResolvedDependency resolvedDependency, Set<String> dependenciesIds) {
-    dependenciesIds.add(getResolvedDependencyId(resolvedDependency));
+    if (!dependenciesIds.add(getResolvedDependencyId(resolvedDependency))) {
+      return;
+    }
 
     if (resolvedDependency.getChildren() != null) {
       for (ResolvedDependency child : resolvedDependency.getChildren()) {

